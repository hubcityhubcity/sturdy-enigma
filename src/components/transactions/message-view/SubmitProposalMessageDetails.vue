<template>
  <div>
    <div class="tx__content__caption">
      <p>
        Submitted {{ transaction.value.proposal_type.toLowerCase() }} proposal
        <b>{{ initialDeposit.amount | atoms | prettyLong }}</b>
        <span>{{ initialDeposit.denom | viewDenom }}</span>
      </p>
    </div>
    <div class="tx__content__information">
      Title
      <i>{{ transaction.value.title }}</i>
    </div>
  </div>
</template>

<script>
import { atoms, viewDenom, prettyLong } from "scripts/num.js"
import { formatBech32 } from "src/filters"

export default {
  name: `submit-proposal-message-details`,
  filters: {
    atoms,
    viewDenom,
    prettyLong,
    formatBech32
  },
  props: {
    transaction: {
      type: Object,
      required: true
    },
    validators: {
      type: Object,
      required: true
    }
  },
  computed: {
    initialDeposit() {
      return this.transaction.value.initial_deposit[0]
    }
  }
}
</script>
