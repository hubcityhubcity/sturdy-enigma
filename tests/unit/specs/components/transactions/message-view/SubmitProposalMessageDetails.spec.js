import { shallowMount } from "@vue/test-utils"
import SubmitProposalMessageDetails from "src/components/transactions/message-view/SubmitProposalMessageDetails"

describe(`SubmitProposalMessageDetails`, () => {
  let wrapper

  // Examples @ https://stargate.cosmos.network/txs?action=submit_proposal
  const tx = {
    height: 2000000,
    value: {
      title: "Bla bla bla",
      description: "Bla bla bla",
      proposal_type: "Text",
      proposer: "cosmos1ey69r37gfxvxg62sh4r0ktpuc46pzjrmz29g45",
      initial_deposit: [
        {
          denom: "uatom",
          amount: "100000000"
        }
      ]
    },
    fee: {
      amount: "37",
      denom: "uatom"
    },
    hash: `ABCD1234`,
    group: `governance`,
    key: "keyhash",
    memo: "(Sent via Lunie)",
    timestamp: 123456789,
    liquidDate: null,
    type: `cosmos-sdk/MsgSubmitProposal`
  }

  it(`renders a submit proposal transaction message`, () => {
    wrapper = shallowMount(SubmitProposalMessageDetails, {
      propsData: {
        transaction: tx,
        validators: {}
      },
      stubs: [`router-link`]
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
