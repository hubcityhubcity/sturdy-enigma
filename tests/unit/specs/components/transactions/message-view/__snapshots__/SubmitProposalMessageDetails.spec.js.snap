// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitProposalMessageDetails renders a submit proposal transaction message 1`] = `
<div
  class="tx__content"
>
  <transactionicon-stub
    transactiongroup="governance"
    transactiontype="Submitted"
  />
   
  <div
    class="tx__content__left"
  >
    <h3>
      Submitted proposal
    </h3>
     
    <span>
      Bla bla bla
    </span>
  </div>
   
  <div
    class="tx__content__right"
  >
    <p
      class="amount"
    >
      
      100Â 
      ATOM
    
    </p>
  </div>
</div>
`;
